src/main/java/net/openhft/chronicle/core/Memory.java

```diff
@@ -164,6 +164,10 @@ default long heapUsed() {
 
     boolean compareAndSwapLong(Object object, long offset, long expected, long value) throws MisAlignedAssertionError;
 
+    int getAndSetInt(long address, int value) throws MisAlignedAssertionError;
+
+    int getAndSetInt(Object object, long offset, int value) throws MisAlignedAssertionError;
```

Comment:
I needed this for `VanillaReferenceCounted#warnAndReleaseIfNotReleased`. Not sure if we want to call this `getAndSet` (consistent with `Unsafe`) or `readAndWrite` (consistent with `Memory`). `compareAndSwap` is consistent with `Unsafe`, so I went with `getAndSet`